redhat/rdma.udev-rules

```diff
@@ -10,3 +10,4 @@ SUBSYSTEM=="module", KERNEL=="mlx*", ACTION=="add", TAG+="systemd", ENV{SYSTEMD_
 SUBSYSTEM=="module", KERNEL=="iw_*", ACTION=="add", TAG+="systemd", ENV{SYSTEMD_WANTS}="rdma.service"
 SUBSYSTEM=="module", KERNEL=="be2net", ACTION=="add", TAG+="systemd", ENV{SYSTEMD_WANTS}="rdma.service"
 SUBSYSTEM=="module", KERNEL=="enic", ACTION=="add", TAG+="systemd", ENV{SYSTEMD_WANTS}="rdma.service"
+SUBSYSTEM=="module", KERNEL=="efa", ACTION=="add", TAG+="systemd", ENV{SYSTEMD_WANTS}="rdma.service"
```

Comment:
Does efa also need the stuff in kernel-boot/ to be updated?

Reply:
I don't think so.

Reply:
I see, this  rules is just badly designed, it should look more like

ACTION=="remove", GOTO="rdma_ulp_modules_end"
SUBSYSTEM!="infiniband", GOTO="rdma_ulp_modules_end"TAG+="systemd", ENV{SYSTEMD_WANTS}+="rdma.service"

Reply:
How do you prefer to handle it then?

Reply:
> I see, this rules is just badly designed, it should look more like
> 
> ACTION=="remove", GOTO="rdma_ulp_modules_end"
> SUBSYSTEM!="infiniband", GOTO="rdma_ulp_modules_end"TAG+="systemd", ENV{SYSTEMD_WANTS}+="rdma.service"

I'm not sure this will have the same functionality, as some of the modules listed in this file are part of the network stack.

Reply:
Left this change untouched for now.

Reply:
The only reason to start rdma.service is if there is a rdma device, so doing it just because a net related module is loaded is already a bug.

RH really needs to fix this stuff. Fedora should be using kernel-boot and it is very unfortunate that RH8 shipped using this junky stuff. @dledford

Reply:
Ok, I'll change this commit to your suggested fix.

Reply:
I suggest to leave it broken and work with RH to get an acceptable outcome, ie use kernel boot or fixup this file.